learn/configuration/instance_options.md

```diff
@@ -4,10 +4,101 @@ sidebarDepth: 2
+You can configure Meilisearch at launch with **command-line options**, **environment variables**, or a **configuration file**.
 
 These startup options affect your entire Meilisearch instance, not just a single index. For settings that affect search within a single index, see [index settings](/learn/configuration/settings.md).
 
+## Configuration file
+
+Meilisearch accepts a configuration file in the `.toml` format as an alternative to command-line options and environment variables. Configuration files can be easily shared and versioned, and allow you to define multiple options.
+
+**Environment variables can overwrite the configuration file, and command-line options can overwrite environment variables.**
+
+You can download the configuration file using the following command:
+
+<CodeSamples id="instance_options_config_file_download_1" />
+
+Once downloaded, Meilisearch will use it as the default configuration file. You will see it when you launch Meilisearch:
+
+```
+888b     d888          d8b 888 d8b                                            888
+8888b   d8888          Y8P 888 Y8P                                            888
+88888b.d88888              888                                                888
+888Y88888P888  .d88b.  888 888 888 .d8888b   .d88b.   8888b.  888d888 .d8888b 88888b.
+888 Y888P 888 d8P  Y8b 888 888 888 88K      d8P  Y8b     "88b 888P"  d88P"    888 "88b
+888  Y8P  888 88888888 888 888 888 "Y8888b. 88888888 .d888888 888    888      888  888
+888   "   888 Y8b.     888 888 888      X88 Y8b.     888  888 888    Y88b.    888  888
+888       888  "Y8888  888 888 888  88888P'  "Y8888  "Y888888 888     "Y8888P 888  888
+
+Config file path:       "./config.toml"
+```
+
+If no configuration file is provided, the `Config file path` is set to `none`.
+
+You can override the default configuration file using the `MEILI_CONFIG_FILE_PATH` environment variable or the `--config-file-path` CLI option.
+
+### Configuration file format
+
+You can configure all environment variables and CLI options using a dedicated key in the configuration file. These keys follow the [snake case](https://en.wikipedia.org/wiki/Snake_case) convention. For example, `--import-dump` must be named `import_dump` in the file, `import_dumps` will throw an error.
+
+::: warning
+The only exception is the `config_file_path` key. Specifying the `config_file_path` key in the configuration file will throw an [error](/learn/configuration/instance_options.md#defining-a-configuration-file-with-a-syntax-error).
+:::
+
+By default, the file is divided into sections separated by comments. Each key has a brief description followed by a link to the relevant section of the documentation. Some keys are commented out as they are not required when launching Meilisearch, for example, `import_snapshot`, or because the value provided in the file is not the default value like `max_indexing_memory`.
+
+### Configuration file errors
+
```

Comment:
Do we really need to define errors in the config file feature? Is it useful for the users? Looks to much like a specification, not a guide.

Reply:
My opinion is the full section can be removed, let me know what you think @maryamsulemani97 and @guimachiavelli

Reply:
I added it here because I though it might be useful, but we can get rid of it

Reply:
From my user POV I would say it's not useful for me to know all the behavior of the features in case of errors. Errors should be explicit enough to guide me, but the docs don't need to describe them. It makes the docs heavier.
But I would rather have the opinion of gui, maybe he does not agree with me.

Reply:
I agree with @curquiza, I also think this section makes the article too long. All the errors seem to work pretty much like other Meilisearch errors, so there's nothing unexpected that justifies talking about them here, I think.